test/__fixtures/BaseTest.sol

```diff
@@ -87,12 +87,13 @@ abstract contract BaseTest is Test {
     function _writeObservation(
         ReservoirPair aPair,
         uint256 aIndex,
-        int112 aPrice,
-        int112 aLiq,
+        int112 aRawPrice,
+        int56 aClampedPrice,
+        int56 aLiq,
         uint32 aTime
     ) internal
     {
-        bytes32 lEncoded = bytes32(abi.encodePacked(aTime, aLiq, aPrice));
+        bytes32 lEncoded = bytes32(abi.encodePacked(aTime, aLiq, aClampedPrice, aRawPrice));
```

Comment:
There's a replacement API for `abi.encodePacked`, which is `bytes.concat`, it could be worth trying out.

Reply:
okay I attempted this. `bytes.concat` needs explicit bytes type and doesn't recognize the int types so we have to cast it manually, sometimes multiple times. Becomes pretty verbose and ugly

So I think no need to use that for now

Reply:
It calls out your struct in code right. uint64 = bytes8, uint128 = bytes16 etc. Right now the reader has no clue whether your bytes32 is going to line up with your abi.encodePacked input

Reply:
My understanding is it should just be `bytes.concat(bytes4(aTime), bytesN(aLiq), bytesN(aClampedPrice), bytesN(aRawPrice))`

Reply:
okay for better readability and verifiability, I've changed it

Reply:
Isn't the conversion unsafe though? `int` -> `uint` -> `bytes` or does solidity not overflow check by default?

Reply:
when we do an explicit conversion in solidity, no overflow check is done. Solidity kind of assumes we know what the hell we're doing when we use explicit conversions

Also I don't know of any other way other than the `int` -> `uint` -> `bytes` path. My research also shows the same